libgraph/include/katana/PropertyGraph.h

```diff
@@ -116,7 +126,28 @@ class KATANA_EXPORT PropertyGraph {
 
   friend class PropertyGraphRetractor;
 
+protected:
+  RDG& rdg() { return *rdg_; }
+  const RDG& rdg() const { return *rdg_; }
+
+  // This constructor is meant to be used by transformation views to share
+  // state with the original graph.
+  PropertyGraph(const PropertyGraph& parent, GraphTopology&& topo) noexcept
```

Comment:
I think this method actually make sense in general, not just for projections. It's just a way to share property data while using a new topology. Please document it as such. I definitely agree it should be protected. If there are additional restrictions (not allowing reordering nodes or whatever), please document those too.